docs/content/integrations/duckdb/using-duckdb-with-dagster.mdx

```diff
@@ -30,45 +29,203 @@ To complete this tutorial, you'll need:
+from dagster_duckdb import DuckDBResource
 
 from dagster import Definitions
 
 defs = Definitions(
     assets=[iris_dataset],
     resources={
-        "io_manager": DuckDBPandasIOManager(
+        "duckdb": DuckDBResource(
             database="path/to/my_duckdb_database.duckdb",  # required
-            schema="IRIS",  # optional, defaults to PUBLIC
         )
     },
 )
+
+
+# end_example
+```
+
+---
+
+### Step 2: Create tables in DuckDB
+
+<TabGroup>
+
+<TabItem name="Create DuckDB tables in Dagster">
+
+#### Create DuckDB tables in Dagster
+
+Using the DuckDB resource, you can create DuckDB tables using the DuckDB Python API:
+
+```python file=/integrations/duckdb/tutorial/resource/create_table.py startafter=start_example endbefore=end_example
+import pandas as pd
+from dagster_duckdb import DuckDBResource
+
+from dagster import asset
+
+
+@asset
+def iris_dataset(duckdb: DuckDBResource) -> None:
+    iris_df = pd.read_csv(
+        "https://docs.dagster.io/assets/iris.csv",
+        names=[
+            "sepal_length_cm",
+            "sepal_width_cm",
+            "petal_length_cm",
+            "petal_width_cm",
+            "species",
+        ],
+    )
+
+    with duckdb.get_connection() as conn:
+        conn.execute("CREATE TABLE iris.iris_dataset AS SELECT * FROM iris_df")
 ```
 
-With this configuration, if you materialized an asset called `iris_dataset`, the DuckDB I/O manager would store the data in the `IRIS.IRIS_DATASET` table in a database stored at `path/to/my_duckdb_database.duckdb`.
+In this example, you define an asset that fetches the Iris dataset as a Pandas DataFrame and renames the columns. Then, using the DuckDB resource, the DataFrame is stored in DuckDB as the `iris.iris_dataset` table.
+
+</TabItem>
+
+<TabItem name="Make existing tables available in Dagster">
+
+#### Make existing tables available in Dagster
 
-Finally, in the <PyObject object="Definitions" /> object, we assign the <PyObject module="dagster_duckdb_pandas" object="DuckDBPandasIOManager" /> to the `io_manager` key. `io_manager` is a reserved key to set the default I/O manager for your assets.
+If you already have tables in DuckDB, you may want to have other assets in Dagster depend on those tables. You can accomplish this by creating [source assets](/concepts/assets/software-defined-assets#defining-external-asset-dependencies) for these tables.
 
-For more info about each of the configuration values, refer to the <PyObject module="dagster_duckdb_pandas" object="DuckDBPandasIOManager" /> API documentation.
+```python file=/integrations/duckdb/tutorial/io_manager/source_asset.py
+from dagster import SourceAsset
+
+iris_harvest_data = SourceAsset(key="iris_harvest_data")
+```
+
+In this example, you create a <PyObject object="SourceAsset" /> for a pre-existing table called `iris_harvest_data`.
+
+</TabItem>
+
+</TabGroup>
+
+
+Now you can run `dagster dev` and materialize the `iris_dataset` asset from the Dagster UI.
 
 ---
```

Comment:
Ugh I can't make a suggestion for this, but no rule is needed here since the following heading is an H3.